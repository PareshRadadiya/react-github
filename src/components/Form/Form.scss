.user-form-container {
	padding-top: 32px;
}

.user-form-logo {
	margin: auto;
	height: 48px !important;
	width: 48px !important;
}

.username-input {
	display: flex;
	justify-content: center;
	align-items: center;
	width: 100%;
	padding: 0;
	margin: 0;

	.username-input-box {
		background: rgb(246, 248, 250);
		border-color: $border-color_1;
		box-shadow: none;
	}
}

.username-input-box {
	padding: 38px;
	max-width: 320px;
	width: 100%;
	border: 1px solid #dbcece;
	border-radius: 20px;
	background-color: $background-color_1;
	fieldset {
		border: 1px solid rgb(216, 222, 228);
	}
	input {
		padding: 12.5px 14px;
		background: #fff;
	}
	label {
		top: -3px;
	}
	label.MuiFormLabel-filled {
		top: 0px;
	}
	label.Mui-focused {
		top: 0px;
	}
	button {
		margin-top: 8px;
	}
}

.submit {
	background-color: $background-color_2 !important;
}

.error {
	color: $color_1;
	font-size: 12px;
	margin-bottom: 5px;
}

.header {
	font-weight: bold;
	letter-spacing: -0.5px;
	text-align: center;
	padding-top: 20px;
	padding-bottom: 20px;
}

.submitButton {
	background-color: $background-color_3 !important;
}
