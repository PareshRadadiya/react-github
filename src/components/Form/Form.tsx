import { useState, FC } from "react";
import { useNavigate } from "react-router-dom";
import {
  Box,
  Avatar,
  Typography,
  TextField,
  Button,
  Paper,
} from "@mui/material";
import gitAvatar from "../../assets/github.png";
import { useTranslation } from 'react-i18next';
import "./Form.scss";

const Form: FC = () => {
  const navigate = useNavigate();
  const [userName, setUserName] = useState<string>("");
  const [error, setError] = useState<boolean>(false);
  const { t } = useTranslation();

  const handleSubmit = () => {
    if (!userName) {
      setError(true);
      return;
    }

    window.localStorage.setItem('userName', userName);
    navigate('/repositories');
  };

  return (
    <>
      <Box className="user-form-container">
        <Avatar alt="Github Avatar" src={gitAvatar} className="user-form-logo" />
        <Typography variant="h5" className="header">
          {t('FORM.GETGITREPO')}
        </Typography>
      </Box>
      <Box className="username-input">
        <Paper className="username-input-box">
          <TextField
            data-testid="user-name"
            fullWidth
            id="outlined-basic"
            label={t('FORM.ENTERNAME')}
            variant="outlined"
            margin="normal"
            value={userName}
            onChange={(e: any) => {
              if (error && e.target.value) {
                setError(false);
              }
              setUserName(e.target.value);
            }}
            error={error}
            required
          />
          {error && <Typography variant="body2" className="error">Please enter github username.</Typography>}
          <Button
            variant="contained"
            className="submitButton"
            fullWidth
            onClick={handleSubmit}
          >
            {t('FORM.SUBMIT')}
          </Button>
        </Paper>
      </Box>
    </>
  );
};

export default Form;
