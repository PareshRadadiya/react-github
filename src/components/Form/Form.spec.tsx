import "@testing-library/jest-dom";
import { render } from '../../tests/test-utils';
import { MemoryRouter } from "react-router-dom";

import Form from "./Form";

describe('Form component', () => {
  it("renders Form correctly", () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={[{ pathname: '/' }]}>
        <Form />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
