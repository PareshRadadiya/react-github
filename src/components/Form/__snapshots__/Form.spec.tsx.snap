// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders Form correctly 1`] = `
<DocumentFragment>
  <div
    class="user-form-container MuiBox-root css-0"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular user-form-logo css-1wlk0hk-MuiAvatar-root"
    >
      <img
        alt="Github Avatar"
        class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
        src="default"
      />
    </div>
    <h5
      class="MuiTypography-root MuiTypography-h5 header css-ag7rrr-MuiTypography-root"
    >
      FORM.GETGITREPO
    </h5>
  </div>
  <div
    class="username-input MuiBox-root css-0"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 username-input-box css-1ps6pg7-MuiPaper-root"
    >
      <div
        class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
        data-testid="user-name"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="outlined-basic"
          id="outlined-basic-label"
        >
          FORM.ENTERNAME
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="outlined-basic"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                FORM.ENTERNAME *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth submitButton css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        FORM.SUBMIT
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
